src/BuildingRegistry.Projections.Extract/BuildingUnitAddressLinkExtract/BuildingUnitAddressLinkExtractProjections.cs

```diff
@@ -58,35 +58,72 @@ await context
+                    message.Message.AddressPersistentLocalId, ct);
             });
 
             When<Envelope<BuildingUnitAddressWasDetachedBecauseAddressWasRemoved>>(async (context, message, ct) =>
             {
-                await RemoveLink(context, message.Message.BuildingUnitPersistentLocalId, message.Message.AddressPersistentLocalId, ct);
+                await RemoveIdempotentLink(context, message.Message.BuildingUnitPersistentLocalId,
+                    message.Message.AddressPersistentLocalId, ct);
             });
 
             When<Envelope<BuildingUnitAddressWasDetachedBecauseAddressWasRejected>>(async (context, message, ct) =>
             {
-                await RemoveLink(context, message.Message.BuildingUnitPersistentLocalId, message.Message.AddressPersistentLocalId, ct);
+                await RemoveIdempotentLink(context, message.Message.BuildingUnitPersistentLocalId,
+                    message.Message.AddressPersistentLocalId, ct);
             });
 
             When<Envelope<BuildingUnitAddressWasDetachedBecauseAddressWasRetired>>(async (context, message, ct) =>
             {
-                await RemoveLink(context, message.Message.BuildingUnitPersistentLocalId, message.Message.AddressPersistentLocalId, ct);
+                await RemoveIdempotentLink(context, message.Message.BuildingUnitPersistentLocalId,
+                    message.Message.AddressPersistentLocalId, ct);
+            });
+
+            When<Envelope<BuildingUnitAddressWasReplacedBecauseAddressWasReaddressed>>(async (context, message, ct) =>
+            {
+                await RemoveIdempotentLink(context, message.Message.BuildingUnitPersistentLocalId,
+                    message.Message.PreviousAddressPersistentLocalId, ct);
+
+                await AddIdempotentLink(context, new BuildingUnitAddressLinkExtractItem
+                {
+                    BuildingPersistentLocalId = message.Message.BuildingPersistentLocalId,
+                    BuildingUnitPersistentLocalId = message.Message.BuildingUnitPersistentLocalId,
+                    AddressPersistentLocalId = message.Message.NewAddressPersistentLocalId,
+                    DbaseRecord = new BuildingUnitAddressLinkDbaseRecord()
+                    {
+                        objecttype = {Value = ObjectType},
+                        adresobjid = {Value = message.Message.BuildingUnitPersistentLocalId.ToString()},
+                        adresid = {Value = message.Message.NewAddressPersistentLocalId}
+                    }.ToBytes(_encoding)
+                }, ct);
             });
         }
 
-        private static async Task RemoveLink(
+        private static async Task AddIdempotentLink(
+            ExtractContext context,
+            BuildingUnitAddressLinkExtractItem linkItem,
+            CancellationToken ct)
+        {
+            var extractItem = await context.BuildingUnitAddressLinkExtract.FindAsync(
```

Comment:
Can we extract the FindAsync method? We both use it in AddIdempotentLink and RemoveIdempotentLink.